pkgs/applications/video/obs-studio/obs-websocket.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, qtbase
+, obs-studio
+, asio_1_10
+, websocketpp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "obs-websocket";
+  version = "4.9.0";
+
+  src = fetchFromGitHub {
+    owner = "Palakis";
+    repo = "obs-websocket";
+    rev = version;
+    sha256 = "1dxih5czcfs1vczbq48784jvmgs8awbsrwk8mdfi4pg8n577cr1w";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ qtbase obs-studio asio_1_10 websocketpp ];
+
+  dontWrapQtApps = true;
+
+  cmakeFlags = with lib; [
+    "-DLIBOBS_INCLUDE_DIR=${obs-studio.src}/libobs"
+  ];
+
+  # obs-studio expects the shared object to be located in bin/32bit or bin/64bit
+  # https://github.com/obsproject/obs-studio/blob/d60c736cb0ec0491013293c8a483d3a6573165cb/libobs/obs-nix.c#L48
+  postInstall = let
+    pluginPath = {
+      i686-linux = "bin/32bit";
+      x86_64-linux = "bin/64bit";
+    }.${stdenv.targetPlatform.system} or (throw "Unsupported system: ${stdenv.targetPlatform.system}");
+  in ''
+    mkdir -p $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
+    ln -s $out/lib/obs-plugins/obs-websocket.so $out/share/obs/obs-plugins/obs-websocket/${pluginPath}
```

Comment:
Maybe move it if it is only required in the target directory.

Reply:
It seems to be required. It is also done by <https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/obs-studio/v4l2sink.nix#L38> and <https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/video/obs-studio/obs-gstreamer.nix#L25>.

Reply:
Should I contact obs-studio Nix package maintainer to have the idea clear on that?

Reply:
This won't be necessary once #125308 is merged. :)

Reply:
Thanks, I will add the plugin to `pkgs/applications/video/obs-studio/plugins/default.nix` once the plugin wrapper is merged :)